Add sorting to customer list

vladPinteaFrontmen opened the pull request:
# Changes
Sorting order applicable to customers list.

Store filters from /customers, /employees and /timesheets

## Related issues
https://github.com/FrontMen/fm-hours/issues/91

## Changed
customers/index.vue Added a select dropdown to manage sort order. Added sorting to computed filteredCustomers
helpers.ts has a generic sortByProp function

## How to test
As an admin, navigate to /customers.
Change sort criteria, note the change in order.
Filter by keyword, note sort order is kept
Selecting "none" will return to unsorted list

Go to /customers, /employees or /timesheets and apply some filter and sort proprieties
Go to a subpage
Return - see the same filters persisting

## Screenshots
![image](https://user-images.githubusercontent.com/85111889/124912946-d72a0d80-dfee-11eb-8eab-79769bb73e1e.png)
![image](https://user-images.githubusercontent.com/85111889/124912967-ddb88500-dfee-11eb-9f65-76db2dd3326f.png)
![image](https://user-images.githubusercontent.com/85111889/124912995-e5782980-dfee-11eb-9d29-d1c488885b9d.png)

Updated images found in the comments

YejunWu168 commented:
Why not base sort on what's selected in the search by field? I think that makes more sense instead of having 4 different options.

vladPinteaFrontmen commented:
![image](https://user-images.githubusercontent.com/85111889/124919376-78689200-dff6-11eb-9198-e5dd563b619a.png)
rebased with Mehmet's changes

vladPinteaFrontmen commented:
> Why not base sort on what's selected in the search by field? I think that makes more sense instead of having 4 different options.

The way I see it, you can still maybe want to see alphabetic order by project for all a debtors projects, for example. In a way the thing that makes more sense is if we filter by debtor we sort by name, and vice versa. 
Additionally we want to know if it's ascending or descnding. 
It further allows us to expand with other proprieties we might want to sort by in the future as the app grows, like say billeable or not, sort archived or not raer than filtering it out, or whatever else. 

Unless the business logic here would be different, I think it's more appropriate to separate the 2.

YejunWu168 commented:
Filters are getting quite big now and a bit confusing. Maybe we should rethink it? Instead of search by you have 
**filter by:** 
 - name
 - debtor
 - archived
 
and then both search and sort will both be based of the **filter by** selection. 
Any thoughts? @mmehmetAliIzci @vladPinteaFrontmen

vladPinteaFrontmen commented:
Added caret on timesheets page 
![image](https://user-images.githubusercontent.com/85111889/125101159-344ebd80-e0da-11eb-9ab3-31cf9014706b.png)

Converted list into a table. Use tablet head to sort the table. Caret is visible. "Archived" is now a toggle, set to false by default. When true, show both archived and not archived customers. 
![image](https://user-images.githubusercontent.com/85111889/125101204-429cd980-e0da-11eb-85fa-62151f2e1ae6.png)

vladPinteaFrontmen commented:
https://user-images.githubusercontent.com/85111889/125419935-b1be4de1-cd6a-4b8c-940b-e76fc555e570.mov
